dat/watchlist.js

```diff
@@ -0,0 +1,82 @@
+const EventEmitter = require('events')
+const emitStream = require('emit-stream')
+
+// dat modules
+const datLibrary = require('../dat/library')
+const watchlistDb = require('../dbs/watchlist')
+
+// globals
+// =
+
+var watchlistEvents = new EventEmitter()
+
+// exported methods
+// =
+
+exports.setup = async function setup () {
+  try {
+    var watchedSites = await watchlistDb.getSites(0)
+    for (let site of watchedSites) {
+      watch(site)
+    }
+  } catch (err) {
+    console.error('Failed to load the watchlist', err)
+  }
+}
+
+exports.addSite = async function addSite(profileId, url, opts) {
+    // validate parameters
+  if (!url || typeof url !== 'string') {
+    console.error('url must be a string')
```

Comment:
We should still throw errors, we just don't need custom error classes.

```suggestion
    throw new Error('url must be a string')
```